### PR TITLE
fix crash when no object to correlate with pixels

### PR DESCRIPTION
**Pull request by londumas**
fix crash when no object to correlate with pixels given the range in r_perp and r_parallel

### CODE REVIEW

File: bin/do_xMetalGrid.py [hunk 1]
```diff
@@ -150,6 +150,10 @@ def corr_func(p):
     pix = healpy.ang2pix(xcf.nside,th,phi)
     print("reading qsos")
 
+    if (ra.size==0):
+        print(" No object to correlate with pixels")
+        sys.exit(1)
```
Comment (ngbusca) [inline bin/do_xMetalGrid.py]: Could you fix the error message to something more clear?

**Comment by londumas** [comment]
"Given the range in rp, there are no objects to correlate with pixels, stop running"

is that better ?

**Comment by ngbusca** [comment]
In fact, why is there a message needed at all? And why is the exit value = 1 (which implies there was an error) ? There's actually no problem, there are 0 pairs within the specified rp, rt, there's nothing special about it.

It's ok that the code exits, but it should result in an empty correlation function.

**Comment by londumas** [comment]
I totally agree.

Currently it tries to read the list of objects and thus crash
sp.amin(zqso)

we can write en empty file if (ra.size==0) and then exit
would that be ok for you ?

**Comment by ngbusca** [comment]
That makes much more sense. It should basically give the same thing as if we had actually run the whole code.

**Comment by londumas** [comment]
ok, I take care of that

**Comment by londumas** [comment]
The last changes work but ~20 lignes, we might think of a nicer and cleaner way to do the same thing

**Comment by londumas** [comment]
I have found it, don't merge yet

**Comment by londumas** [comment]
@ngbusca this solution is cleaner and works very well

**Pull request merged**